app/scripts/lib/PatchStore.ts

```diff
@@ -100,4 +117,26 @@ export class PatchStore {
       })
       .filter(Boolean) as Patch[];
   }
+
+  private _normalizeEventPatches(
```

Comment:
Nit: A doc comment would be really helpful here. It took a while to understand the intent here.

I think what it's meant to do is ensure all patches have a path length of at least 1. Is that correct?

Reply:
Done. That's correct, it is "rebasing" root level patches to apply to the entire state to avoid removing all the state from the other controllers.